lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -78,54 +78,62 @@ def self.deliver(*args)
     begin
       miq_task&.state_active
       object_name = "#{options[:object_type]}.#{options[:object_id]}"
-      _log.info("Delivering #{ManageIQ::Password.sanitize_string(options[:attrs].inspect)} for object [#{object_name}] with state [#{state}] to Automate")
       automate_attrs = automate_attrs_from_options(options)
-
       if options[:object_type]
         vmdb_object = options[:object_type].constantize.find_by!(:id => options[:object_id])
         automate_attrs[create_automation_attribute_key(vmdb_object)] = options[:object_id]
         vmdb_object.before_ae_starts(options) if vmdb_object.respond_to?(:before_ae_starts)
         vmdb_object.mark_execution_servers if vmdb_object.respond_to?(:mark_execution_servers)
       end
-
       uri = create_automation_object(options[:instance_name], automate_attrs, create_automation_object_options(options, vmdb_object))
       ws  = resolve_automation_object(uri, user_obj)
+      miq_request_id = nil
+
+      if !defined?(ws.root.attributes).nil?
+        if !ws.root.attributes['miq_request_id'].nil?
+          miq_request_id = ws.root.attributes['miq_request_id']
+        elsif !defined?(ws.root.attributes[ws.root.attributes['vmdb_object_type']].miq_request_id).nil?
+          miq_request_id = ws.root.attributes[ws.root.attributes['vmdb_object_type']].miq_request_id
+        elsif !defined?(ws.root.attributes[ws.root.attributes['vmdb_object_type']].object.miq_request_id).nil?
+          miq_request_id = ws.root.attributes[ws.root.attributes['vmdb_object_type']].object.miq_request_id
+        end
+      end
+
+      attr_string = ManageIQ::Password.sanitize_string(options[:attrs].inspect)
+      $request_log.request_info("Delivering #{attr_string} for object [#{object_name}] with state [#{state}] to Automate", {:resource_id => miq_request_id})
 
       if ws.nil? || ws.root.nil?
-        message = "Error delivering #{ManageIQ::Password.sanitize_string(options[:attrs].inspect)} for object [#{object_name}] with state [#{state}] to Automate: Empty Workspace"
-        _log.error(message)
+        message = "Error delivering #{attr_string} for object [#{object_name}] with state [#{state}] to Automate: Empty Workspace"
+        $request_log.request_error(message, {:resource_id => miq_request_id})
         return nil
       end
-
```

Comment:
Why were these blank lines removed? Thinking we should leave the original code as untouched as possible.